### ISSUE TITLE
Ergonomics - update .set()

### ISSUE THREAD

**Issue by senseibaka** [neuledge/ddb-table#157]
Hi,

First off I just wanted to say that I really like using this library, it just feels like the right balance of complexity and ease of use. It's much easier to use than plain aws sdk calls, while still letting me understand what will be sent to the dynamodb endpoint.

Anyway, sometimes in my code, I write something like:
```
const update = table.update(keys)
  .set('x', somethingDefined);

if (couldBeUndefined !== undefined) {
  update.set('y', couldBeUndefined)
}

await update.exec();
```

When really I'd prefer to write something like:
```
await table.update(keys)
  .set('x', somethingDefined)
  .set('y', couldBeUndefined)
  .exec();
```

But, if I were to do it the second way, I'd get back an error like:
> Invalid UpdateExpression: An expression attribute value used in expression is not defined; attribute value: y

---

I was wondering if `.set()` could either:
* Ignore `undefined` values by default, or
* Perhaps offer an optional configuration, with a default behaviour of passing through the value to dynamodb if not specified (for backwards compat), e.g.
```
.set(attribute, value, opts?: { ifUndefined?: 'remove' | 'ignore' });
```

Thoughts?
Cheers

**Comment by moshest** [comment]
What's your type definition? Did you defined `y` as optional?

```ts
interface MyDocument {
 x: TypeOfX;
 y?: TypeOfY;
}

**Comment by senseibaka** [comment]
Yeah that's correct, `y` is optional.

**Comment by moshest** [comment]
I believe in this case, the default behavior should be setting it as undefined, ie. unsetting the field.
I will see if I can push a fix in the next few days.

**Comment by moshest** [comment]
released as `v2.1.0`
